pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java

```diff
@@ -62,7 +62,7 @@ public enum AggregationFunctionType {
   SUMINT("sumInt", ReturnTypes.AGG_SUM, OperandTypes.INTEGER),
   SUMLONG("sumLong", ReturnTypes.AGG_SUM, OperandTypes.or(OperandTypes.INTEGER, OperandTypes.ARRAY_OF_INTEGER)),
   SUMPRECISION("sumPrecision", ReturnTypes.explicit(SqlTypeName.DECIMAL), OperandTypes.ANY, SqlTypeName.OTHER),
-  AVG("avg", SqlTypeName.OTHER, SqlTypeName.DOUBLE),
+  AVG("avg", ReturnTypes.DOUBLE, OperandTypes.or(OperandTypes.NUMERIC, OperandTypes.ARRAY), SqlTypeName.OTHER),
```

Comment:
```suggestion
  AVG("avg", ReturnTypes.DOUBLE, OperandTypes.or(OperandTypes.NUMERIC, OperandTypes.ARRAY), SqlTypeName.OTHER, SqlTypeName.DOUBLE),
```

We should retain the final return type